projects/framework-blender/source/ftrack_framework_blender/__init__.py

```diff
@@ -0,0 +1,265 @@
+import platform
+
+from PySide6 import QtWidgets
+
+import bpy
+
+from ftrack_framework_core.host import Host
+from ftrack_framework_core.event import EventManager
+from ftrack_framework_core.client import Client
+from ftrack_framework_core.registry import Registry
+from ftrack_framework_core.configure_logging import configure_logging
+
+from ftrack_constants import framework as constants
+
+from ftrack_utils.extensions.environment import (
+    get_extensions_path_from_environment,
+)
+from ftrack_utils.usage import set_usage_tracker, UsageTracker
+
+from ftrack_utils.session import create_api_session
+
+
+# Evaluate version and log package version
+try:
+    from ftrack_utils.version import get_version
+
+    __version__ = get_version(
+        os.path.basename(os.path.dirname(__file__)),
+        os.path.dirname(os.path.dirname(os.path.dirname(__file__))),
+    )
+except Exception:
+    __version__ = '0.0.0'
+
+
+configure_logging(
+    'ftrack_framework_blender',
+    extra_modules=['ftrack_qt'],
+    propagate=False,
+)
+
+logger = logging.getLogger(__name__)
+logger.debug('v{}'.format(__version__))
+
+app = QtWidgets.QApplication.instance()
+if app is None:
+    app = QtWidgets.QApplication([])
+
+on_menu_tools = []
+
+
+# Operator to execute commands
+class FTRACK_OT_ExecuteCommand(bpy.types.Operator):
+    bl_idname = "ftrack.execute_command"
+    bl_label = "Execute Command"
+
+    func_key: bpy.props.StringProperty()
+
+    def execute(self, context):
+        try:
+            for tool in on_menu_tools:
+                if tool.get("label") == self.func_key:
+                    func = tool.get("command")
+                    func()
+        except Exception as e:
+            self.report({'ERROR'}, f"Error executing command: {e}")
+            return {'CANCELLED'}
+
+        # Register a timer to process Qt events
+        bpy.app.timers.register(self.process_qt_events)
+        return {'FINISHED'}
+
+    def process_qt_events(self):
+        app.processEvents()
+        return 0.01  # Call this function every 0.01 seconds
+
+
+def create_ftrack_menu(tools_list):
+    class FTRACK_MT_Menu(bpy.types.Menu):
+        bl_label = "ftrack"
+        bl_idname = "FTRACK_MT_menu"
+
+        def draw(self, context):
+            layout = self.layout
+            for tool in tools_list:
+                op = layout.operator(
+                    "ftrack.execute_command", text=tool.get("label")
+                )
+                op.func_key = tool.get("label")
+
+    return FTRACK_MT_Menu
+
+
+# Function to add the menu to the UI
+def draw_ftrack_menu(self, context):
+    self.layout.menu("FTRACK_MT_menu")
+
+
+# We could use the ftrack @invoke_in_qt_main_thread utility which should act
+# the same way as the DCC specific one. But using this means it ensures
+# synchronization with Blender’s undo stack, event loop, and other core
+# functionalities, which might not be guaranteed if you use Qt’s threading
+# mechanisms directly.
```

Comment:
I think we should use the invoke_in_qt_main_thread for UI always so we prevent unexpected crashes.

Reply:
I'm checking but doesn't seems Blender has anything like Maya .... :/ 
I'll remove the whole util folder with the rest of the code for now.